.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,60 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
We will need some staging environment for the Docker images so we have more control over publishing.
If all the latest main code goes directly into a public image I'm a bit concerned there could be some mistakes. 
Wdyt?